plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryStorageArrowPageSource.java

```diff
@@ -135,8 +139,8 @@ public Page getNextPage()
     public long getMemoryUsage()
     {
         long memoryUsage = streamBufferAllocator.getAllocatedMemory();
-        if (split.getDataSize().isPresent()) {
-            memoryUsage += split.getDataSize().getAsInt() + pageBuilder.getSizeInBytes();
+        if (splitDataSize.isPresent()) {
```

Comment:
we don't need splitDataSize as we don't hold the split anymore
you can add size of `streamName` to `memoryUsage` if that matters

Reply:
🤦 fixed!